ethcore/light/src/on_demand/mod.rs

```diff
@@ -94,6 +94,25 @@ pub mod error {
 	}
 }
 
+/// Public interface for performing network requests `OnDemand`
+pub trait OnDemandRequester: Send + Sync {
+	/// Submit a strongly-typed batch of requests.
+	///
+	/// Fails if back-reference are not coherent.
+	fn request<T>(&self, ctx: &BasicContext, requests: T) -> Result<OnResponses<T>, basic_request::NoSuchOutput>
+	where
+		T: request::RequestAdapter,
+		Self: Sized;
```

Comment:
I wonder why `Self: Sized` is required?

Reply:
Good question. Prevents being used as a trait object?

Reply:
But this method is generic, I thought it's not allowed in a trait object already.

Reply:
Thanks guys, it was not needed :+1: :1st_place_medal: